plugins/jobs/src/js/components/form/VolumesFormSection.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from "react";
+import { Trans } from "@lingui/macro";
+import { Tooltip } from "reactjs-components";
+import { MountService } from "foundation-ui";
+
+import AddButton from "#SRC/js/components/form/AddButton";
+import FieldError from "#SRC/js/components/form/FieldError";
+import FieldInput from "#SRC/js/components/form/FieldInput";
+import FieldLabel from "#SRC/js/components/form/FieldLabel";
+import FieldSelect from "#SRC/js/components/form/FieldSelect";
+import FormGroup from "#SRC/js/components/form/FormGroup";
+import DeleteRowButton from "#SRC/js/components/form/DeleteRowButton";
+import FormGroupHeading from "#SRC/js/components/form/FormGroupHeading";
+import FormGroupHeadingContent from "#SRC/js/components/form/FormGroupHeadingContent";
+import FormRow from "#SRC/js/components/form/FormRow";
+import InfoTooltipIcon from "#SRC/js/components/form/InfoTooltipIcon";
+import MetadataStore from "#SRC/js/stores/MetadataStore";
+import {
+  FormOutput,
+  FormError,
+  JobVolume,
+  SecretVolume,
+  VolumeMode
+} from "./helpers/JobFormData";
+import { getFieldError } from "./helpers/ErrorUtil";
+
+interface VolumesSectionProps {
+  formData: FormOutput;
+  errors: FormError[];
+  showErrors: boolean;
+  onRemoveItem: (path: string, index: number) => void;
+  onAddItem: (path: string) => void;
+}
+
+function isFBS(volume: JobVolume | SecretVolume): boolean {
+  return volume.hasOwnProperty("secret");
+}
+
+class VolumesFormSection extends React.Component<VolumesSectionProps> {
+  constructor(props: VolumesSectionProps) {
+    super(props);
+  }
+
+  getVolumesLines() {
+    const {
+      formData: { volumes = [] },
+      onRemoveItem,
+      errors,
+      showErrors
+    } = this.props;
+
+    return volumes.map((volume: JobVolume | SecretVolume, index: number) => {
+      if (isFBS(volume)) {
```

Comment:
Does this indicate SecretVolume? I think it must since everything gets cast as JobVolumn below. I wonder if it would be better to filter out all the SecretVolumns before the map and then avoid all the casting later? Your index === 0 thing would also not be a bug (if I understand this correctly)

```
volumes.filter((volume: JobVolume | SecretVolume) => isFBS(volume)
       .map((volume: JobVolume) => { ... }
```

Reply:
If I filter the list the error indices may not match up properly in the UI - but I changed the return type on `isFBS` so that the type casting is not necessary anymore!